dulwich/client.py

```diff
@@ -57,6 +57,8 @@
 except ImportError:
     import urllib.parse as urlparse
 
+import urllib3
```

Comment:
Please don't import urllib3 as a top-level import - both for performance reason, but also to allow people to use dulwich without having urllib3 installed if they don't need http support.

Reply:
Makes sense. Fixed.